### PR TITLE
Create new Serial GCC7.2.0 PR build with C++14 set

### LINKED ISSUE #8071: PR Testing: Create a new GCC 7.2.0 C++14 Serial PR build
## Enhancement
@trilinos/framework 

We are looking to add -std=c++14 to a GCC 7.2.0 build. While some packages are using C++14 (and C++17) features, this will move us in the right direction and ensure that customers will be able to build Trilinos with C++14.
Moving to C++14 will also require us to turn off testing for the 4.9.3 Serial build, so this GCC 7.2.0 build will also be a serial build.

This incremental step is consistent with discussions at the Trilinos Product Leaders meeting and is related to #6260.

### PR DESCRIPTION
**Pull request by ZUUL42**
@trilinos/framework 

## Motivation

This will add the needed code for a new Serial GCC 7.2.0 C++14 PR build.
This addresses and closes #8071 

## Stakeholder Feedback

This is work moves to a configuration that is c++14 ready making the PR testing scheme ready for Kokkos 3.3 to be merged in.
## Testing

A new Jenkins job was created and tested building with this new code.
[CDash](https://testing.sandia.gov/cdash/index.php?project=Trilinos&display=project&filtercount=3&showfilters=1&filtercombine=and&field1=site&compare1=61&value1=ascic158&field2=buildname&compare2=61&value2=PR-gcc72c14serial-test-Trilinos_pullrequest_gcc_7.2.0_serial-1&field3=buildstamp&compare3=61&value3=20200922-1624-Experimental)

## Additional Information
A few WError flags had to be turned off to get a clean build.
```
-Wno-unused-but-set-variable
-Wno-unused-variable
-Wno-unused-label
```
This shouldn't be an issue since these warning are still protected against with the GCC 8.3.0 [Parallel] PR build.

### CODE REVIEW

File: cmake/std/PullRequestLinuxGCC7.2.0SerialTestingSettings.cmake [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# Misc options typically added by CI testing mode in TriBITS
+
+# Use the below option only when submitting to the dashboard
+#set (CTEST_USE_LAUNCHERS ON CACHE BOOL "Set by default for PR testing")
+
+# Disabling fortran for this build as that is a customer use case
+set (Trilinos_ENABLE_Fortran OFF CACHE BOOL "Set by default for SERIAL PR test")
+
+set (MPI_EXEC_PRE_NUMPROCS_FLAGS "--bind-to;none" CACHE STRING "Set by default for PR testing")
```
Comment (jwillenbring) [inline cmake/std/PullRequestLinuxGCC7.2.0SerialTestingSettings.cmake:19]: Why is this needed for a serial build?

File: cmake/std/PullRequestLinuxGCC7.2.0SerialTestingSettings.cmake [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# NOTE: The above is a workaround for the problem of having threads on MPI
+# ranks bind to the same cores (see #2422).
+
+set (Trilinos_ENABLE_COMPLEX_DOUBLE ON CACHE BOOL "Set by default for PR testing to exercise complex doubles case")
+
+# Disable just one Teko sub-unit test that fails with openmpi 1.10 (#2712)
+set (Teko_DISABLE_LSCSTABALIZED_TPETRA_ALPAH_INV_D ON CACHE BOOL "Temporarily disabled in PR testing")
```
Comment (jwillenbring) [inline cmake/std/PullRequestLinuxGCC7.2.0SerialTestingSettings.cmake:26]: We should be able to turn this test back on for a serial build if this is an openmpi issue.

File: cmake/std/PullRequestLinuxGCC7.2.0SerialTestingSettings.cmake (hunk 1 above)
Comment (ZUUL42) [reply cmake/std/PullRequestLinuxGCC7.2.0SerialTestingSettings.cmake:19]: 👍 I'll remove the line.

File: cmake/std/PullRequestLinuxGCC7.2.0SerialTestingSettings.cmake (hunk 2 above)
Comment (ZUUL42) [reply cmake/std/PullRequestLinuxGCC7.2.0SerialTestingSettings.cmake:26]: 👍 I'll remove the line.

**Review by prwolfe — APPROVED** [review]
Looks right to me.
